packages/core/src/render3/component_ref.ts

```diff
@@ -118,8 +118,26 @@ export class ComponentFactory<T> extends AbstractComponentFactory<T> {
   override ngContentSelectors: string[];
   isBoundToModule: boolean;
 
-  override get inputs(): {propName: string; templateName: string;}[] {
-    return toRefArray(this.componentDef.inputs);
+  override get inputs(): {
+    propName: string,
+    templateName: string,
+    transform?: (value: any) => any,
+  }[] {
+    const refArray = toRefArray(this.componentDef.inputs) as {
+      propName: string,
+      templateName: string,
+      transform?: (value: any) => any,
+    }[];
+
+    if (this.componentDef.inputTransforms !== null) {
+      for (const input of refArray) {
+        if (this.componentDef.inputTransforms.hasOwnProperty(input.propName)) {
+          input.transform = this.componentDef.inputTransforms[input.propName];
```

Comment:
Just a tiny refactoring proposal for a better minification (`this.componentDef.inputTransforms` would not get minified, so we can avoid referencing it multiple times): 
```suggestion
    const componentDef = this.componentDef;
    const inputTransforms = componentDef.inputTransforms;
    const refArray = toRefArray(componentDef.inputs) as {
      propName: string,
      templateName: string,
      transform?: (value: any) => any,
    }[];

    if (inputTransforms !== null) {
      for (const input of refArray) {
        if (inputTransforms.hasOwnProperty(input.propName)) {
          input.transform = inputTransforms[input.propName];
```